client/reducers/events.js

```diff
@@ -116,14 +116,39 @@ const eventsReducer = createReducer(initialState, {
             selectedEvent: payload,
         }
     ),
-    [EVENTS.ACTIONS.OPEN_EVENT_DETAILS]: (state, payload) => (
-        {
-            ...state,
-            showEventDetails: payload,
-            highlightedEvent: payload,
-            readOnly: false,
+    [EVENTS.ACTIONS.OPEN_EVENT_DETAILS]: (state, payload) => {
+        if (payload.lock_user) {
+            let _events = cloneDeep(state.events)
+            let event = payload
+
+            // Change dates to moment objects
```

Comment:
This is needed when the back-end returns the event after locking it as the dates won't be moment objects. I will have the lock/unlock action update the state and see if this can be removed.